[wxMSW] Deprecate wxBitmap::SetWidth/SetHeight etc. functions

a-wi opened the pull request:
- Implement new SetHBITMAP overloaded function which sets HBITMAP together with its parameters (size and colour depth).
- Implement new SetHICON overloaded function which sets HICON together with its parameters (size).
- Deprecate SetWidth, SetHeight, SetSize, SetDepth and legacy SetHBITMAP, SetHICON functions.
- Refactor the code in order to not use deprecated method.

vadz commented on include/wx/msw/bitmap.h:
```diff
@@ -199,8 +199,11 @@ class WXDLLIMPEXP_CORE wxBitmap : public wxGDIImage,
+    wxDEPRECATED_INLINE(void SetHBITMAP(WXHBITMAP bmp), SetHandle((WXHANDLE)bmp); )
+#endif // WXWIN_COMPATIBILITY_3_0
     WXHBITMAP GetHBITMAP() const { return (WXHBITMAP)GetHandle(); }
+    void SetHBITMAP(WXHBITMAP bmp, int width, int height, int depth);
```
This is minor, but I think it might be better to call this function `InitHBITMAP()` (or maybe `InitFromHBITMAP()`?) rather than `Set` for 2 reasons: first, to avoid any confusion with the deprecated overload and second, to indicate that it can/should be used with the yet otherwise non-initialized bitmap.

vadz commented on include/wx/msw/icon.h:
```diff
@@ -69,8 +69,12 @@ class WXDLLIMPEXP_CORE wxIcon : public wxGDIImage
+#endif // WXWIN_COMPATIBILITY_3_0
+
     WXHICON GetHICON() const { return (WXHICON)GetHandle(); }
+    void SetHICON(WXHICON icon, int width, int height);
```
Of course, if we rename the other method to `InitHBITMAP()`, this one should be renamed ot `InitHICON()` too for consistency.

vadz commented on src/msw/bitmap.cpp:
```diff
@@ -1290,6 +1290,32 @@ void wxBitmap::SetMask(wxMask *mask)
+
+    AllocExclusive();
+
+    GetBitmapData()->m_handle =(WXHANDLE)bmp;
```
Nitpick: missing space after `=`

vadz commented on src/msw/bitmap.cpp:
```diff
@@ -1290,6 +1290,32 @@ void wxBitmap::SetMask(wxMask *mask)
 
+void wxBitmap::SetHBITMAP(WXHBITMAP bmp, int width, int height, int depth)
+{
+#if wxDEBUG_LEVEL
```
Is `::GetObject()` really fast/free? If not, I'd make this check `wxDEBUG_LEVEL >= 2` i.e. avoid always paying its price in the default build with `wxDEBUG_LEVEL == 1` as we don't want to pessimize bitmap code, this is one of the places where the speed does matter.

vadz reviewed (approved):
Thanks for cleaning this up!

I globally think this is fine, but, in addition to the already mentioned trivial renaming, I wonder if we might want to do the following:

- Change `InitFromHXXX()` to return true/false depending on whether the handle is valid or not -- it looks like calling them is often followed by `if ( IsOk() )` tests, so this might make the code more compact.
- Add a separate `ResetHXXX()` which might be a tad more clear than `InitFromHXXX(0, 0, 0)`.

But it's up to you to decide if you think this would be useful, of course.

Thanks again!

a-wi committed:
Set HBITMAP and its parameters in one call

To avoid separate calls to SetWidth/Height/Size/Depth functions after calling SetHBITMAP() use newly implemented InitFromHBITMAP() function which allows to set HBITMAP together with its parameters in one call.

a-wi committed:
Set HICON and its parameters in one call

To avoid separate calls to SetSize() function after calling SetHICON() use newly implemented InitFromHICON() function which allows set HICON together with its parameters in one call.

a-wi committed:
Deprecate wxBitmap::SetWidth/Height/Size/Depth, wxIcon::SetSize functions

wxBitmap and wxIcon dimensions and colour depth are determined during creation so using all these functions could lead to inconsistencies.

a-wi committed:
Don't use deprecated SetHBITMAP() method

Inside wxBitmap we have a direct access to wxBitampData so HBITMAP can be stored directly in this structure and there is no need to call SetHBITMAP() function.
